frontend/src/pages/Home/UserGuestListPage.js

```diff
@@ -0,0 +1,377 @@
+                        }}
+                    >
+                        Remove
+                    </Button>,
+                ]),
+                // Add a row to add a guest
+                [
+                    <>
+                        <Label for="last-name" hidden={true}>
+                            Last name
+                        </Label>
+                        <Field
+                            type="text"
+                            maxLength="255"
+                            required
+                            name="lastName"
+                            component={CustomInputForm}
+                        />
+                    </>,
+                    <>
+                        <Label for="first-name" hidden={true}>
+                            First name
+                        </Label>
+                        <Field
+                            type="text"
+                            maxLength="255"
+                            required
+                            name="firstName"
+                            component={CustomInputForm}
+                        />
+                    </>,
+                    <Button type="submit">Add</Button>,
+                ],
+            ];
+        }
+
+        // Get the guests
+        axios.get(`/api/users/${props.user.pk}/guest_list/`).then((res) => {
+            setGuestList(guestListToTable(res.data.guest_list));
+            setOneTimeEvents(
+                res.data.one_time_events.map((event) => {
+                    return {
+                        name: event.name,
+                        startTime: event.start_time,
+                        endTime: event.end_time,
+                        guestList: guestListToTable(event.guest_list),
+                        pk: event.pk,
+                    };
+                })
+            );
+        });
+        setLoading(false);
+    }, [loading, props.user.pk]);
+
+    function onRenewalModeSubmit(values) {
+        setRenewalModeStatus("Updating...");
+        axios
+            .patch(`/api/users/${props.user.pk}/`, {
+                guest_list_renewal_mode: values.renewalMode,
+            })
+            .then(() => {
+                setRenewalModeStatus("Updated!");
+                setRenewalMode(values.renewalMode);
+                setLoading(true);
+                // Remove the "Updated!" thing after 2 seconds
+                setTimeout(() => setRenewalModeStatus(""), 2000);
+            });
+    }
+
+    function onGuestRemoval(guest) {
+        // TODO: Use a modal for confirmation?
+        if (
+            window.confirm(
+                `Remove ${guest.first_name} ${guest.last_name} from your guest list?`
+            )
+        ) {
+            axios.delete(`/api/guests/${guest.pk}/`).then(() => {
+                setLoading(true);
+            });
+        }
+    }
+
+    function onEventDeletion(event) {
+        // TODO: Use a modal for confirmation?
```

Comment:
I don't think a modal is necessary since the actions are not critical (they can just add them to the guest list again). It probably would be a nice addition, though idk how well it would be if they wanted to bulk delete.